__tests__/mockClient.test.js

```diff
@@ -0,0 +1,96 @@
+const Q = require('q');
+const request = Q.denodeify(require('request'));
+const MockClient = require('../src/client/Client');
+const MOCKSERVER_URI = 'localhost:3000';
+
+describe('when adding routes on the same uri', () => {
+	let mockClient;
+	beforeEach(() => {
+		mockClient = new MockClient(3030, MOCKSERVER_URI);
+	});
+
+	it('should return first response', done => {
+		mockClient.addRoute({
+			uri: '/duplicate-route',
+			method: 'GET',
+			response: {
+				body: 'First response'
+			}
+		})
+		.then(() => {
+			return request('http://localhost:3030/duplicate-route').spread((response, body) => {
+				expect(response.statusCode).toBe(200);
+				expect(body).toBe('First response');
+			});
+		})
+		.then(done, done.fail);
+	});
+
+	it('should return second response', done => {
+		mockClient.addRoute({
+			uri: '/duplicate-route',
+			method: 'GET',
+			response: {
+				body: 'Second response'
+			}
+		})
+		.then(() => {
+			return request('http://localhost:3030/duplicate-route').spread((response, body) => {
+				expect(response.statusCode).toBe(200);
+				expect(body).toBe('Second response');
+			});
+		})
+		.then(done, done.fail);
+	});
+});
+
+describe('clean', () => {
+	let mockClient;
+
+	beforeAll(() => {
+		mockClient = new MockClient(2020, MOCKSERVER_URI);
+	});
+
+	it('should be possible to add a route and call it', done => {
+		// Add route
+		mockClient.addRoute({
+			uri: '/test',
+			method: 'get',
+			response: {
+				body: 'hello world'
+			}
+		})
+
+		// Route can be called
+		.then(() => {
+			return request('http://localhost:2020/test').spread((response, body) => {
+				expect(response.statusCode).toBe(200);
+				expect(body).toBe('hello world');
+			});
+		})
+		.then(done, done.fail);
+	});
+
+	it('should not be possible to call the route after it has been cleared', done => {
+		mockClient.clear()
+
+		// requests are cleared
+		.then(() => mockClient.getRequests())
+		.then(res => {
+			expect(res).toEqual([]);
+		})
+
+		// Route can no longer be called
+		.then(() => {
+			request('http://localhost:2020/test')
+				.then(done.fail)
+				.catch(() => {});
+		})
+		.then(done, done.fail);
+	});
+});
+
+afterEach(() => {
+	const mockClient = new MockClient(3030, MOCKSERVER_URI);
```

Comment:
Ah. Sorry. I thought it was the one inside the `description`.

Now I understand why you put it here. It is actually to cleanup for the first description. But you cannot put it inside since we need the mock client not to be cleared after the first `it`.

The hierarchy looks a bit strange but I understand your point. If we have to do this, I'm fine with it.